icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java

```diff
@@ -6613,6 +6614,25 @@ public void test20348_CurrencyPrefixOverride() {
             "$100.00", fmt.format(100));
     }
 
+    @Test
+    public void test20956_MonetarySymbolGetters() {
+        Locale locale = new Locale.Builder().setLocale(Locale.forLanguageTag("et")).build();
+        DecimalFormat decimalFormat = (DecimalFormat) NumberFormat.getCurrencyInstance(locale);
+        Currency currency = Currency.getInstance("EEK");
+
+        decimalFormat.setCurrency(currency);
+
+        DecimalFormatSymbols decimalFormatSymbols = decimalFormat.getDecimalFormatSymbols();
+        assertEquals("MONETARY DECIMAL SEPARATOR", ".", decimalFormatSymbols.getMonetaryDecimalSeparatorString());
+        assertEquals("DECIMAL SEPARATOR", ",", decimalFormatSymbols.getDecimalSeparatorString());
+        assertEquals("MONETARY GROUPING SEPARATOR", " ", decimalFormatSymbols.getMonetaryGroupingSeparatorString());
+        assertEquals("GROUPING SEPARATOR", " ", decimalFormatSymbols.getGroupingSeparatorString());
+        assertEquals("CURRENCY SYMBOL", "kr", decimalFormatSymbols.getCurrencySymbol());
+
+        StringBuffer string = decimalFormat.format(new BigDecimal(12345.12), new StringBuffer(), DontCareFieldPosition.INSTANCE);
```

Comment:
Change variable `string` to something else?  The call in the following line of `string.toString()` seems strange.  Other instances in the file use `buf`, `buffer`, etc. (ex: `pat`).

Reply:
Yeah, I had copied and pasted this test from the bug thread.  Cleaned up.